src/_pytest/main.py

```diff
@@ -491,14 +491,24 @@ def gethookproxy(self, fspath):
             proxy = self.config.hook
         return proxy
 
+    _deselected = []
```

Comment:
this **must** be a instance variable instead of a class one

Reply:
Yeah.  As you can see from its placement this is just a hack / proof of concept anyway.